### DISCUSSION TITLE
Automate documentation in regards to validators

### DISCUSSION THREAD

**Discussion by LongnoseRob** [pymeasure/pymeasure#731] [Ideas]
In many properties and methods in new instrument implementations we are using `valdiators` and `values`.
Would it be possible by means of sphinx to automate the documentation, so that we would not need to mention the "allowed/valid range" manually in the docstring?

```
    frequency = Instrument.setting(
        "FR %9.0f HZ",
        """
        controls the output frequency of the instrument
        For the 8567B the valid range is 100 kHz to 2060 MHz
        """,
        validator=strict_range,
        values=[1.0E5, 2.060E9],
        )
```

I did a little search but did not find an quick answer how we could trigger adding a text block to the propert/method documentation if the source mentions `validator `or `values`.

But maybe someone with more experience with sphinx and template modification has some ideas?

**Comment by bilderbuchi** [comment]
Yes, that should be possible to automate during property creation. See #571.

**Reply by LongnoseRob** [reply]
Oh thats already onthe agenda, soll close this one here..
